crates/flux-refineck/src/constraint_gen.rs

```diff
@@ -146,6 +147,26 @@ impl<'a, 'tcx> ConstrGen<'a, 'tcx> {
         Ok(res)
     }
 
+    fn check_generic_args(
+        &self,
+        did: DefId,
+        generic_args: &[GenericArg],
+    ) -> Result<(), CheckerErrKind> {
+        let generics = self.genv.generics_of(did)?;
+        for (idx, arg) in generic_args.iter().enumerate() {
+            let param = generics.param_at(idx, self.genv)?;
+            match param.kind {
+                GenericParamDefKind::SplTy | GenericParamDefKind::BaseTy => {
```

Comment:
This feels weird. If the param is `GenericParamDefKind::BaseTy` we should only be checking that the argument is `GenericArg::BaseTy`. We only need to go into the type for parameters of kind `SplTy`

Reply:
restricted the check to only `BaseTy`